src/app/dynamic-forms/fields/rich-text/rich-text.component.ts

```diff
@@ -310,6 +342,86 @@ export class RichTextComponent implements OnDestroy, AfterViewInit {
     }
   }
 
+  /**
+   * Insert Image 
+   */
+  
+  insertImage(): void{
+
+      const selection = this.editor.view.state.selection;
+      const selectedText = selection ? this.editor.state.doc.textBetween(selection.from, selection.to) : undefined;
+
+      const dialogRef: MatDialogRef<AddRichTextImageComponent> = this._dialog.open(
+        AddRichTextImageComponent,
+        {
+          data: {
+            href: this.imageURL == "" ? selectedText : this.imageURL
+          },
+          maxHeight: "80vh",
+          width: "600px"
+        }
+      );
+
+      dialogRef.afterClosed().subscribe({
```

Comment:
should this subscribe method be here in the insertImage() method? if you subscribe each time this method is called, would there be an accumulation of many subscriptions and callbacks for the dialogRef.afterClosed() event?

Reply:
I believe that is a self-completing observable, so there's no memory accumulation.

https://stackoverflow.com/questions/58198544/angular-dialogref-unsubscribe-do-i-need-to-unsubscribe-from-afterclosed

Reply:
Interesting, thanks! Makes sense.